kroxylicious/src/main/java/io/kroxylicious/proxy/internal/DefaultFilterContext.java

```diff
@@ -213,9 +264,14 @@ public void forwardResponse(ApiMessage response) {
     public void closeConnection() {
         this.channelContext.close().addListener(future -> {
             LOGGER.debug("{} closed.", channelDescriptor());
+            closeFuture.complete(null);
         });
     }
 
+    public void discard() {
```

Comment:
Added this for the case where we:
1. from a request filter, use `sendRequest` to send an out-of-band-request
2. do not forward the original request
3. when the out-of-band response completion stage completes, forward a response to the client

Because we need a way to tell the FilterHandler that the request has been handled and we can move on to the next one.

I'm thinking instead we should pass a KrpcFilterContext to the completion stage, so it would become:

`CompletionStage<(KrpcFilterContext, ApiMessage)> sendRequest(....)`

Then we would have a separate context we could await that's more particular to the out-of-band response.